hamilton/io/materialization.py

```diff
@@ -195,6 +197,85 @@ class to(Materialize):
+    ):
+        """Modifies the signature to include the parameters from *all* dataclasses.
+        Note this just replaces **kwargs with the union of the parameters. Its not
+        strictly correct, as (a) its a superset of the available ones and (b) it doesn't
+        include the source/value parameters. However, this can help the development experience
+        on jupyter notebooks, and is a good enough approximation for now.
+
+
+        :param fn: Function to modify -- will change the signature.
+        :param dataclasses_union: All dataclasses to union.
+        :return: The function without **kwargs and with the union of the parameters.
+        """
+        original_signature = inspect.signature(fn)
+        original_parameters = list(original_signature.parameters.values())
+
+        new_parameters = []
+        seen = set()
+        for dataclass in dataclasses_union:
+            for field in dataclasses.fields(dataclass):
+                if field.name not in seen:
+                    new_parameters.append(
+                        inspect.Parameter(
+                            field.name,
+                            inspect.Parameter.KEYWORD_ONLY,
+                            annotation=field.type,
+                            default=None,
+                        )
+                    )
+                seen.add(field.name)
+
+        # Combining old and new parameters
+        # Checking for position of **kwargs and insert new params before
+        for idx, param in enumerate(original_parameters):
+            if param.kind == inspect.Parameter.VAR_KEYWORD:
+                break
+        else:
+            idx = len(original_parameters)
+
+        # Insert new parameters while respecting the order
+        combined_parameters = original_parameters[:idx] + new_parameters + original_parameters[idx:]
+        combined_parameters = [param for param in combined_parameters if param.name != "kwargs"]
+
+        # Creating a new signature with combined parameters
+        new_signature = original_signature.replace(parameters=combined_parameters)
+
+        # Creating a new function with the new signature
+        @functools.wraps(fn)
+        def wrapper(*args, **kwargs):
+            bound_arguments = new_signature.bind(*args, **kwargs)
+            return fn(*bound_arguments.args, **bound_arguments.kwargs)
+
+        # Assign the new signature to the wrapper function
+        wrapper.__signature__ = new_signature
+        wrapper.__doc__ = f"""
+        Materializes data to {key} format. Note that the parameters are a superset of possible parameters -- this might depend on
+        the actual type of the data passed in. For more information, see: https://hamilton.dagworks.io/en/latest/reference/io/available-data-adapters/#data-loaders.
+        You can also pass `source` and `value` in as kwargs.
+        """
+        return wrapper
+
+    for key, item in SAVER_REGISTRY.items():
```

Comment:
what about loaders?

Reply:
Materializará don't need loaders. If we add `.inject` we will.